hta/common/trace.py

```diff
@@ -415,14 +525,73 @@ def _get_profiler_step(ts: int) -> int:
     return profiler_steps
 
 
+# @profile
+def parse_trace_dataframe_json(
+    trace_file_path: str, cfg: ParserConfig
+) -> Tuple[MetaData, pd.DataFrame, TraceSymbolTable]:
+    trace_record = parse_trace_dict(trace_file_path)
+    meta: Dict[str, Any] = {k: v for k, v in trace_record.items() if k != "traceEvents"}
+    df: pd.DataFrame = pd.DataFrame()
+    if "traceEvents" in trace_record:
+        df = pd.DataFrame(trace_record["traceEvents"])
+
+        # assign an index to each event
+        df.reset_index(inplace=True)
+        df["index"] = pd.to_numeric(df["index"], downcast="integer")
+
+        df, local_symbol_table = compress_df(df, cfg)
+
+    return meta, df, local_symbol_table
+
+
+# @profile
+def parse_trace_dataframe_ijson(
+    trace_file_path: str, cfg: ParserConfig
+) -> Tuple[MetaData, pd.DataFrame, TraceSymbolTable]:
+    # TODO print backend
+    meta: Dict[str, Any] = {}
+    # k: v for k, v in trace_record.items() if k != "traceEvents"}
+
+    df = parse_trace_events_ijson(trace_file_path)
+
+    # assign an index to each event
+    df.reset_index(inplace=True)
+    df["index"] = pd.to_numeric(df["index"], downcast="integer")
+
+    df, local_symbol_table = compress_df(df, cfg)
+    return meta, df, local_symbol_table
+
+
+# @profile
+def parse_trace_dataframe_ijson_batched(
+    trace_file_path: str, cfg: ParserConfig, compress_on_fly: bool = False
+) -> Tuple[MetaData, pd.DataFrame, TraceSymbolTable]:
+    # TODO print backend
+    meta: Dict[str, Any] = {}
+    # k: v for k, v in trace_record.items() if k != "traceEvents"}
+
+    df = parse_trace_events_ijson_batched(trace_file_path, cfg, compress_on_fly)
+
+    # assign an index to each event
+    df.reset_index(inplace=True)
+    df["index"] = pd.to_numeric(df["index"], downcast="integer")
+
+    df, local_symbol_table = compress_df(df, cfg)
+    return meta, df, local_symbol_table
+
+
 def parse_trace_dataframe(
-    trace_file_path: str, cfg: Optional[ParserConfig] = None
+    trace_file_path: str,
+    cfg: Optional[ParserConfig] = None,
+    parser_backend: str = get_trace_parsing_backend(),
+    trace_memory=False,
```

Comment:
Since we already have ParserConfig, the code would be cleaner if we add trace_memory and parser_backend to the ParserConfig argument.